### PR TITLE
[refactor](Nereids) Refactor all rewrite logical unit tests by match-pattern

### LINKED ISSUE #17625: [Enhancement] refactor `Nereids Unit Test` by match-pattern
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Currently, EliminateFilterTest directly check PlanTree by checking Memo.

Use MemoPatternMatchSupported to match result.

A example is  `ExtractFilterFromCrossJoinTest`

- [ ] MergeFiltersTest
- [ ] EliminateLimitTest 
- [ ] EliminateFilterTest
- [ ] SplitLimitTest
- [ ] FindHashConditionForJoinTest

### Solution

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by reswqa**
# Proposed changes

Issue Number: close #17625

## Problem summary

Refactor all rewrite logical unit tests by match-pattern.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by reswqa** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.42 seconds
 stream load tsv:          463 seconds loaded 74807831229 Bytes, about 154 MB/s
 stream load json:         40 seconds loaded 2358488459 Bytes, about 56 MB/s
 stream load orc:          75 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230311160627_clickbench_pr_112828.html

**Comment by englefly** [comment]
In my understanding, this pr is not about "enhancement". 
Could you title it with "codeStyle" or something else, please.
"enhancement" is specific to performance improvement.

File: fe/fe-core/src/test/java/org/apache/doris/nereids/util/PlanChecker.java [hunk 1]
```diff
@@ -126,6 +127,12 @@ public PlanChecker applyTopDown(RuleFactory ruleFactory) {
         return applyTopDown(ruleFactory.buildRules());
     }
 
+    public PlanChecker applyTopDown(CustomRewriter customRewriter) {
```
Comment (jackwener) [inline fe/fe-core/src/test/java/org/apache/doris/nereids/util/PlanChecker.java:130]: 👍

File: fe/fe-core/src/test/java/org/apache/doris/nereids/rules/rewrite/logical/EliminateUnnecessaryProjectTest.java [hunk 2]
```diff
@@ -55,57 +52,49 @@ protected void runBeforeAll() throws Exception {
...
                 .build();
 
-        CascadesContext cascadesContext = MemoTestUtils.createCascadesContext(unnecessaryProject);
-        cascadesContext.topDownRewrite(new EliminateUnnecessaryProject());
-
-        Plan actual = cascadesContext.getMemo().copyOut();
-        Assertions.assertTrue(actual.child(0) instanceof LogicalProject);
+        PlanChecker.from(MemoTestUtils.createConnectContext(), unnecessaryProject)
+                .applyTopDown(new EliminateUnnecessaryProject())
+                .matches(logicalProject());
```
Comment (jackwener) [inline fe/fe-core/src/test/java/org/apache/doris/nereids/rules/rewrite/logical/EliminateUnnecessaryProjectTest.java:63]: please use matchesFromRoot in this test class

**Review by jackwener — COMMENTED** [review]
LGTM except a small suggestion

**Comment by reswqa** [comment]
Thanks @englefly for the remind and @jackwener for the reveiw, I have updated this commit's prefix to `Refactor` and address your comments, please take a look again.

**Comment by reswqa** [comment]
run buildall

**Review by jackwener — APPROVED** [review]

**Pull request merged**